kmip/core/factories/attribute_values.py

```diff
@@ -40,184 +23,138 @@ class AttributeValueFactory(object):
-        elif name is AttributeType.LEASE_TIME:
-            value = self._create_lease_time(value)
-        elif name is AttributeType.USAGE_LIMITS:
-            value = self._create_usage_limits(value)
-        elif name is AttributeType.STATE:
-            value = self._create_state(value)
-        elif name is AttributeType.INITIAL_DATE:
-            value = self._create_initial_date(value)
-        elif name is AttributeType.ACTIVATION_DATE:
-            value = self._create_activation_date(value)
-        elif name is AttributeType.PROCESS_START_DATE:
-            value = self._create_process_start_date(value)
-        elif name is AttributeType.PROTECT_STOP_DATE:
-            value = self._create_protect_stop_date(value)
-        elif name is AttributeType.DEACTIVATION_DATE:
-            value = self._create_deactivation_date(value)
-        elif name is AttributeType.DESTROY_DATE:
-            value = self._create_destroy_date(value)
-        elif name is AttributeType.COMPROMISE_OCCURRENCE_DATE:
-            value = self._create_compromise_occurrence_date(value)
-        elif name is AttributeType.COMPROMISE_DATE:
-            value = self._create_compromise_date(value)
-        elif name is AttributeType.REVOCATION_REASON:
-            value = self._create_revocation_reason(value)
-        elif name is AttributeType.ARCHIVE_DATE:
-            value = self._create_archive_date(value)
-        elif name is AttributeType.OBJECT_GROUP:
-            value = self._create_object_group(value)
-        elif name is AttributeType.FRESH:
-            value = self._create_fresh(value)
-        elif name is AttributeType.LINK:
-            value = self._create_link(value)
-        elif name is AttributeType.APPLICATION_SPECIFIC_INFORMATION:
-            value = self._create_application_specific_information(value)
-        elif name is AttributeType.CONTACT_INFORMATION:
-            value = self._create_contact_information(value)
-        elif name is AttributeType.LAST_CHANGE_DATE:
-            value = self._create_last_change_date(value)
-        elif name is AttributeType.CUSTOM_ATTRIBUTE:
-            value = self._create_custom_attribute(value)
+        if name is enums.AttributeType.UNIQUE_IDENTIFIER:
+            return attributes.UniqueIdentifier(value)
+        elif name is enums.AttributeType.NAME:
+            return self._create_name(value)
+        elif name is enums.AttributeType.OBJECT_TYPE:
+            return attributes.ObjectType()
+        elif name is enums.AttributeType.CRYPTOGRAPHIC_ALGORITHM:
+            return attributes.CryptographicAlgorithm(value)
+        elif name is enums.AttributeType.CRYPTOGRAPHIC_LENGTH:
+            return self._create_cryptographic_length(value)
+        elif name is enums.AttributeType.CRYPTOGRAPHIC_PARAMETERS:
+            return self._create_cryptographic_parameters(value)
+        elif name is enums.AttributeType.CRYPTOGRAPHIC_DOMAIN_PARAMETERS:
+            raise NotImplementedError()
```

Comment:
There are quite a few of the attributes that contained a "create" function within them and now raise a "NotImplementedError" error. Why the change?